### ISSUE TITLE
Re-quadrangulation, or "as a level designer I have little use for triangles"

### ISSUE THREAD

**Issue by joewintergreen** [gabrielw-us/ProBuilder-Unreal-Public-Tracker#72]
From a level design perspective - putting together rooms, etc - faces becoming triangulated is a pain and not really something that is ever useful (except in the case of an actual triangular room, or more complex geo, etc). 

I understand that stuff has to get triangulated at some stage (and that's obviously fine) but from a usability perspective, it would be better if things were visualised wherever possible as quads. I am not sure whether I am articulating this well. 
One way of putting it is, yeah, any square floor ends up being two or more triangles, but as a level designer I have no use for this information. It's a "probuilder as level design tool" versus "probuilder as traditional modeling tool" question, I guess.

I think it would be cool to automatically re-quadrangulate faces which became triangulated as a result of vertex manipulation "bending the face" or whatever, but which don't need to be triangulated anymore.

**Comment by arthursw** [comment]
You can already merge faces for that purpose. Doing it automatically is doable, but I am not sure it's an important feature for now...

I am not sure people would bend a face, put it back flat without using the undo command (which would remerge faces).

**Comment by gabrielw-us** [comment]
Agreed with @arthursw . Doing this would be risky, would make major changes to the mesh without asking the user. Can't distinguish well enough between tris/ngons the user wants, and those they'd rather be quads. The "Merge Faces" tool does this nicely, if needed.

**Comment by arthursw** [comment]
We could have a setting "automatically merge faces when they become coplanar" in a far future.

**Comment by joewintergreen** [comment]
I Want That

*Joe Wintergreen*
*Director*
*Impromptu Games* <http://impromptugames.com/>
*facebook <http://www.facebook.com/ImpromptuGames> twitter
<https://twitter.com/joewintergreen> linkedin
<http://au.linkedin.com/in/joewintergreenarthur>*
